kiwi_lint/similar_string.py

```diff
@@ -0,0 +1,70 @@
+import string
+
+import astroid
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+from textdistance import levenshtein
+
+
+class SimilarStringChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "similar-string-checker"
+
+    msgs = {
+        "R5011": (
+            'A similar string to "%s" exists: "%s"',
+            "similar-string",
+            "Similar strings should be avoided for single translation",
+        )
+    }
+
+    def __init__(self, linter):
+        super().__init__(linter)
+        self._dict_of_strings = {}
+        self.threshold = 0.8
+
+    @staticmethod
+    def clean_string(text):
+        for word in text:
+            if word not in string.punctuation:
+                text += "".join(word)
+        return text
+
+    @staticmethod
+    def marked_for_translation(funcname):
+        return funcname in ("_", "gettext_lazy")
+
+    def check_similar_string(self, translation_string):
+        for key in self._dict_of_strings:
+            if (
+                levenshtein.normalized_similarity(
+                    self.clean_string(translation_string), self.clean_string(key)
+                )
+                >= self.threshold
+            ):
+                return key
+        return None
+
+    def check_same_string(self, translation_string):
+        return translation_string in self._dict_of_strings
```

Comment:
method used only in 1 place. Remove the method and use the condition directly. It makes the visit_call method more readable.